### PR TITLE
feat: added filters

### PR DESCRIPTION
**Pull request by PeeyushPrashant**
- filter by price
- filter by category
- filter by rating
- sorting by price

Screenshot (65)

### CODE REVIEW

File: src/components/Aside/Aside.jsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import "./Aside.css"
+import{useFilter} from "../../hooks/useFilter";
+const Aside= ()=>{
```
Comment (anuj-kumary) [inline src/components/Aside/Aside.jsx]: Give one line space between your import and your components function.

File: src/components/ProductCard/ProductCard.css [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+.product-title {
+  margin: 0;
+}
+.product-rating {
+  padding-right: 1.5rem;
+  color: black;
+  font-size: 1.5rem;
+}
+.price-tag {
+  color: black;
```
Comment (anuj-kumary) [inline src/components/ProductCard/ProductCard.css]: Don't hardcode color name use as a variable.

File: src/components/ProductCard/ProductCard.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+const ProductCard=({image,title,price,description,rating})=>{
+return (
+    
+          <div class="card relative">
+            <div>
+              <section className="relative">
+                  <div className="absolute wishlist-icon "><i class="far fa-heart icon-sm"></i></div>
+                <img src={image} alt="sneaker" className="card-img"/>
+              </section>
+              <section class="sec-2">
```
Comment (anuj-kumary) [inline src/components/ProductCard/ProductCard.jsx]: change class to className

File: src/contexts/Filter-context.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { createContext, useReducer } from "react";
+import { filterReducer } from "../reducer/filterReducer";
+const FilterContext = createContext();
+
+var initialFilter;
```
Comment (anuj-kumary) [inline src/contexts/Filter-context.js:5]: Why you're using **` var`** here.

File: src/pages/ProductList/ProductList.jsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                <Aside/>
+                <div class="product-container">
+                  {filteredProduct.map((item)=>{
+                      return (
+                          <ProductCard
+                          image={item.image}
+                          title={item.title}
+                         price={item.price}
+                         description={item.description}
+                         rating={item.rating}
```
Comment (anuj-kumary) [inline src/pages/ProductList/ProductList.jsx]: You can use here Destructuring so need to repeat item.

**Review by anuj-kumary — APPROVED** [review]
Looks very clean code
Suggestion:-
- In some places, you forget to change class to className
- Give some space between importing your components and function

File: src/components/Aside/Aside.jsx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+  const {filterstate,dispatch} = useFilter();
+    return (
+        <aside className="filters flex-col">
+          <div className="sub-section flex-row">
+            <h3 className="heading">Filters</h3>
+            <p className="clear-out"
+            onClick={()=>dispatch({type:"clear"})}
```
Comment (rohit8686) [inline src/components/Aside/Aside.jsx:11]: you can give dispatch type in uppercase

**Review by rohit8686 — APPROVED** [review]

File: src/pages/ProductList/ProductList.jsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    const {state}= useProductList();
+    const {filterstate}= useFilter();
+    const initialProduct= state.initialProduct;
+
+    const union=(...arr)=>{
+      const concatedProducts= arr.reduce((acc,curr)=>acc.concat(curr));
+      return concatedProducts;
+    }
+
+    const categoryFilter= (product,category)=>{
```
Comment (vinc3nati) [inline src/pages/ProductList/ProductList.jsx:17]: good use of filter

**Review by vinc3nati — APPROVED** [review]

File: src/contexts/Filter-context.js (hunk 4 above)
Comment (PeeyushPrashant) [reply src/contexts/Filter-context.js:5]: because I am passing this object to my reducer function in some other file

**Pull request merged**